meson.build

```diff
@@ -7,11 +7,39 @@ compiler = meson.get_compiler('cpp')
 
 static_deps = get_option('static-linkage') or get_option('default_library') == 'static'
 
-# See https://github.com/kiwix/libkiwix/issues/371
-if ['arm', 'mips', 'm68k', 'ppc', 'sh4'].contains(host_machine.cpu_family())
-  extra_libs = ['-latomic']
+# Atomics as compiled by GCC or clang can lead to external references to
+# functions depending on the type size and the platform. LLVM provides them in
+# 'libcompiler_rt', which clang normally automatically links in, while GNU
+# provides them in 'libatomic', which GCC *does not* link in automatically (but
+# this is probably going to change, see
+# https://gcc.gnu.org/bugzilla/show_bug.cgi?id=81358). Regardless of the setup
+# of the compiler driver itself (GCC or clang), we can thus assume that if some
+# atomic references can't be resolved, then 'libatomic' is missing.
+atomics_program = '''
+#include <atomic>
+#include <cstdint>
+
+using namespace std;
+
+int main() {
+  volatile atomic_bool a_b = true;
+  volatile atomic_ullong a_ull = -1;
+  // Next two lines are to cover atomic<socket_t> from 'httplib.h'.
+  volatile atomic<uint32_t> a_u32 = -1;
+  volatile atomic<uint64_t> a_u64 = -1;
+
+  return atomic_load(&a_b) == false && atomic_load(&a_ull) == 0 &&
+    atomic_load(&a_u32) == 0 && atomic_load(&a_u64) == 0;
+}
+'''
+if compiler.links(atomics_program,
+                  name: 'compiler driver readily supports atomics')
+	extra_libs = []
 else
-  extra_libs = []
+  libatomic = compiler.find_library('atomic')
+  compiler.links(atomics_program, name: 'atomics work with libatomic',
+                 dependencies: libatomic, required: true)
+  extra_libs = ['-latomic']
 endif
```

Comment:
I know that this is the first step in building the `extra_libs` collection. However since it has become much more verbose now, I would prefer that `extra_libs = []` is moved upward outside the `if` (and probably placed under the definition of `compiler`) and the inclusion of `libatomic` in `extra_libs` is performed via `+=` similar to how that collection is populated with other libs. Then this code will be safer to shuffle around.